includes/modules/additional_images.php

```diff
@@ -24,75 +24,87 @@
 
 // do not check for additional images when turned off
 if ($products_image !== '' && !empty($flag_show_product_info_additional_images)) {
-    // prepare image name
-    $products_image_extension = substr($products_image, strrpos($products_image, '.'));
-    $products_image_base = str_replace($products_image_extension, '', $products_image);
-
-    // if in a subdirectory
-    if (strrpos($products_image, '/')) {
-        $products_image_match = substr($products_image, strrpos($products_image, '/') + 1);
-        //echo 'TEST 1: I match ' . $products_image_match . ' - ' . $file . ' -  base ' . $products_image_base . '<br>';
-        $products_image_match = str_replace($products_image_extension, '', $products_image_match) . '_';
-        $products_image_base = $products_image_match;
-    }
+    if (ADDITIONAL_IMAGES_APPROACH === 'modern') {
+        $images_array = [];
+        $sql = "SELECT additional_image FROM " . TABLE_PRODUCTS_ADDITIONAL_IMAGES . " WHERE products_id = :productsID";
+        $sql = $db->bindVars($sql, ':productsID', (int)$_GET['products_id'], 'integer');
+        $images_query = $db->Execute($sql);
+
+        while (!$images_query->EOF) {
+            $images_array[] = DIR_WS_IMAGES . $images_query->fields['additional_image'];
+            $images_query->MoveNext();
```

Comment:
Using `foreach` instead of `while ! EOF` would eliminate the need for the `->MoveNext()` call.

In lots of places.